ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,568 @@
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"abort_incomplete_multipart_upload": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"days_after_initiation": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"expiration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validate.ValidateUTCFormat,
+									},
+									"days": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validate.ValidateAllowedRangeInt(1, 3650),
+										Default:      0, // API returns 0
+									},
+									"expired_object_delete_marker": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"filter": {
+							Type:             schema.TypeList,
+							Required:         true,
+							DiffSuppressFunc: suppressMissingFilterConfigurationBlock,
+							// IBM has filter a required parameter
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"prefix": {
+										Type:     schema.TypeString, // check if prefix empty is eccepted and if filter empty is accepted
+										Optional: true,
+									},
+								},
+							},
+						},
+						"rule_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 255), // check with team if we have any validation function
+						},
+						"noncurrent_version_expiration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"noncurrent_days": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validation.IntAtLeast(1),
+									},
+								},
+							},
+						},
+						"status": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{"Enabled", "Disabled"}),
+						},
+						"transition": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validate.ValidateUTCFormat,
+									},
+									"days": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validate.ValidateAllowedRangeInt(0, 3650),
```

Comment:
Is it 3650 correct or a typo?

Reply:
Yes it is correct , I have referenced this from our existing schema and used that validation itself.